dtfs-central-api/src/services/state-machines/utilisation-report/event-handlers/delete-payment/delete-payment.event-handler.test.ts

```diff
@@ -0,0 +1,321 @@
+import { EntityManager } from 'typeorm';
+import { FeeRecordEntity, FeeRecordEntityMockBuilder, PaymentEntity, PaymentEntityMockBuilder, UtilisationReportEntityMockBuilder } from '@ukef/dtfs2-common';
+import { when } from 'jest-when';
+import { handleUtilisationReportDeletePaymentEvent } from './delete-payment.event-handler';
+import { NotFoundError } from '../../../../../errors';
+import { FeeRecordStateMachine } from '../../../fee-record/fee-record.state-machine';
+import { aDbRequestSource } from '../../../../../../test-helpers/test-data/db-request-source';
+import { feeRecordsAndPaymentsMatch } from '../../../../../helpers/fee-record-matching';
+import { FeeRecordPaymentDeletedEvent } from '../../../fee-record/event-handlers';
+
+jest.mock('../../../../../helpers/fee-record-matching');
+
+describe('handleUtilisationReportPaymentDeletedEvent', () => {
+  const mockRemove = jest.fn();
+  const mockFindOne = jest.fn();
+  const mockEntityManager = {
+    remove: mockRemove,
+    findOne: mockFindOne,
+  } as unknown as EntityManager;
+
+  const aMockEventHandler = () => jest.fn();
+  const aMockFeeRecordStateMachine = (eventHandler: jest.Mock): FeeRecordStateMachine =>
+    ({
+      handleEvent: eventHandler,
+    }) as unknown as FeeRecordStateMachine;
+
+  beforeEach(() => {
+    jest.spyOn(FeeRecordStateMachine, 'forFeeRecord').mockReturnValue(aMockFeeRecordStateMachine(aMockEventHandler()));
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks();
+    jest.restoreAllMocks();
+  });
+
+  it('throws a NotFoundError if there is no payment for that payment id', async () => {
+    // Arrange
+    const report = UtilisationReportEntityMockBuilder.forStatus('RECONCILIATION_IN_PROGRESS').build();
+    when(mockFindOne).calledWith(PaymentEntity, expect.anything()).mockResolvedValue(null);
+
+    // Act / Assert
+    await expect(
+      handleUtilisationReportDeletePaymentEvent(report, {
+        transactionEntityManager: mockEntityManager,
+        paymentId: 1,
+        requestSource: aDbRequestSource(),
+      }),
+    ).rejects.toThrow(NotFoundError);
+  });
+
+  it('removes payment if there are no linked fee records', async () => {
```

Comment:
I spoke to Josh briefly about this, I agree with having this in but it did make me think:
The ticket that has us editing what fee records go with a payment - we should check if payments can be 'orphaned' and unattached from all fee records and how we would then find them (e.g. the payments tab if it's going to exist on the individual bank report page)

Reply:
Ooh interesting, yeah I wonder if we also want to link payments directly to a report...